agent/engine/docker_container_engine.go

```diff
@@ -160,6 +162,55 @@ func (dg *dockerGoClient) dockerClient() (dockeriface.Client, error) {
 	return dg.clientFactory.GetClient(dg.version)
 }
 
+func (dg *dockerGoClient) LoadImage(name string, reader io.ReadCloser) DockerContainerMetadata {
+	timeout := ttime.After(loadImageTimeout)
+	pullLock.Lock()
+	defer pullLock.Unlock()
+
+	response := make(chan DockerContainerMetadata, 1)
+
+	go func() { response <- dg.loadImage(name, reader) }()
+
+	select {
+	case resp := <-response:
+		return resp
+	case <-timeout:
+		return DockerContainerMetadata{Error: &DockerTimeoutError{loadImageTimeout, "loaded"}}
+	}
+}
+
+func (dg *dockerGoClient) loadImage(name string, reader io.ReadCloser) DockerContainerMetadata {
+	log.Debug("Loading image", "name", name)
+	defer reader.Close()
+	client, err := dg.dockerClient()
+	if err != nil {
+		return DockerContainerMetadata{Error: CannotGetDockerClientError{version: dg.version, err: err}}
+	}
+
+	_, err = client.InspectImage(name)
```

Comment:
That makes sense. We currently push to a different object on every deploy, so this isn't an issue for us. ETag based sounds much better (and more general).